File: 2024/June/6-4 409 Longest Palindrome/LongestPalindrome.java
class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character, Integer> counts = new HashMap<Character, Integer>();
        int oddCount = 0;

        for(char c : s.toCharArray()) {
            counts.put(c, counts.getOrDefault(c, 0) + 1);
            if(counts.get(c) % 2 == 0) {
                oddCount--;
            }
            else {
                oddCount++;
            }
        }

        if(oddCount > 1) {
            return s.length() - oddCount + 1;
        }
        return s.length();
    }
}

File: 2024/June/6-4 409 Longest Palindrome/LongestPalindrome.go
package main

func main() {

}

func longestPalindrome(s string) int {
	var counts = make(map[rune]int)
	var oddCount = 0

	for _, c := range s {
		counts[c] = counts[c] + 1
		if counts[c]%2 == 0 {
			oddCount--
		} else {
			oddCount++
		}
	}

	if oddCount > 1 {
		return len(s) - oddCount + 1
	}
	return len(s)
}
